R/Rules-class.R

```diff
@@ -2057,53 +2057,46 @@ CohortSizeConst <- function(size) {
   .CohortSizeConst(size = safeInteger(size))
 }
 
-# nolint start
+# CohortSizeParts ----
 
-## --------------------------------------------------
-## Cohort size based on the parts
-## --------------------------------------------------
-
-##' Cohort size based on the parts
-##'
-##' This class is used when the cohort size should change for the second part of
-##' the dose escalation. Only works in conjunction with
-##' \code{\linkS4class{DataParts}} objects.
-##'
-##' @slot sizes the two sizes for part 1 and part 2
-##'
-##' @keywords classes
-##' @example examples/Rules-class-CohortSizeParts.R
-##' @export
-.CohortSizeParts <-
-    setClass(Class="CohortSizeParts",
-             representation(sizes="integer"),
-             prototype(sizes=as.integer(c(1, 3))),
-             contains="CohortSize",
-             validity=
-                 function(object){
-                     o <- Validate()
+## class ----
 
-                     o$check(all(object@sizes > 0),
-                             "the cohort sizes need to be positive")
-                     o$check(identical(length(object@sizes), 2L),
-                             "2 elements required in sizes")
+#' `CohortSizeParts`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`CohortSizeParts`] is the class for cohort size that changes for the second
+#' part of the dose escalation. It works only in conjunction with [`DataParts`]
+#' objects.
+#'
+#' @slot sizes (`integer`)\cr a vector of length two with two sizes, one for
```

Comment:
```suggestion
#' @slot sizes (`numeric`)\cr a vector of length two with two sizes, one for
```
?

Reply:
Hi @danielinteractive - why `numeric`? According to NEST data types, `integer` is ok.

```
numeric | numeric vector (integer or double)
number | numeric scalar (integer or double)
proportion | number between 0 and 1 (exclusive)
integer | integer vector
```

Reply:
alright